pkg/lib/cockpit-components-file-autocomplete.jsx

```diff
@@ -61,9 +61,13 @@ export class FileAutoComplete extends React.Component {
                     .find(entry => (entry.type == 'directory' && entry.path == path + '/') || (entry.type == 'file' && entry.path == path));
 
             if (match) {
+                // If match file path is a prefix of another file, do not update current directory,
+                // since we cannot tell file/directory user wants to select
+                // https://bugzilla.redhat.com/show_bug.cgi?id=2097662
+                const isPrefix = this.state.displayFiles.filter(entry => entry.path.startsWith(value)).length > 1;
```

Comment:
it's cleaner to write it like:
const isPrefix = this.state.displayFiles.some(entry => entry.path.startsWith(value);

Reply:
But here I'm checking if the number of results matches IS HIGHER THAN 1 (e.g. for input /tmp/dir you get 2 matches: /tmp/dir/ and /tmp/dir-1/).

Since the input will always match with itself, what you wrote will always be true.

Reply:
Oh true - sorry - wrong suggestion :)